lib/fetching/content-services/facebook-content-service.js

```diff
@@ -9,133 +44,108 @@ var _ = require('underscore');
+  if (source == null) {
+    process.nextTick(function() {
+      self.emit('error', new Error('Invalid URL'));
+    });
+    callback([]);
+    return;
+  }
 
-  this._doRequest(queries, function(err, res) {
+  this._doRequest(source, function(err, data) {
     if (err) {
-      self.emit('error', new Error(err.message))
+      self.emit('error', new Error(err.message));
       return callback([]);
     }
-
-    // This will be an array of info about the source.
-    var sourceInfo = _.findWhere(res.data, {name: 'source'}).fql_result_set;
-
-    if (sourceInfo.length === 0) {
-      self.emit('error', new Error('Invalid Facebook URL'));
-      callback([]);
-    } else {
-      var posts = _.findWhere(res.data, {name: 'posts'}).fql_result_set;
-      var comments = _.findWhere(res.data, {name: 'comments'}).fql_result_set;
-      var authors = _.findWhere(res.data, {name: 'authors'}).fql_result_set;
-      callback(self._handleResults(posts, comments, authors));
-    }
+    callback(self._handleResults(data));
   });
 };
 
-// Runs the FQL request
-// queries - The queries to run
-// callback(err, data) - The callback via which to return the resulting data.
-FacebookContentService.prototype._doRequest = function(queries, callback) {
-  graph.fql(queries, function(err, res) {
-    if (err)
+FacebookContentService.prototype._getSourceFromUrl = function(url) {
+  var URLREGEX = new RegExp(this._urlRegex);
+
+  return URLREGEX.exec(url)[0];
+};
+
+FacebookContentService.prototype._doRequest = function(source, callback) {
+  graph.get(source + '/feed', this._getParams, function(err, res) {
+    if (err) {
       callback(err, []);
-    else
-      callback(null, res);
+    } else if (res.data && res.data.length > 0) {
+      callback(err, res.data);
+    }
   });
 };
 
-// Handle data returned by the FQL query. Returns an array of hashes of report data
-// (does not need to be sorted, parent class handles that).
-FacebookContentService.prototype._handleResults = function(posts, comments, authors) {
+FacebookContentService.prototype._handleResults = function(data) {
   var self = this;
   var reportData = [];
-  var postsById = _.indexBy(posts, 'post_id');
 
-  // Handle each post.
-  posts.forEach(function(post) {
-    reportData.push(self._parse(post, {type: 'post', authors: authors}));
-  });
+  if (!data) {
+    return reportData;
+  }
 
-  // Handle each comment.
-  comments.forEach(function(comment) {
-    reportData.push(self._parse(comment, {type: 'comment', authors: authors, post: postsById[comment.post_id]}));
+  data.forEach(function(post) {
+      // Handle each post and any comments already attached to it
+      reportData.push(self._parse(post));
+      if (post.comments && post.comments.data && post.comments.data.length > 0) {
+        post.comments.data.forEach(function(comment) {
+          comment.link = post.link;
+          reportData.push(self._parse(comment));
+        });
+      }
```

Comment:
I think generally map + concat should be preferred to each + push, as it's faster and IMO cleaner. Not that performance really matters here though. Also if we have underscore around we might as well use it.